freqtrade/freqtradebot.py

```diff
@@ -606,29 +606,31 @@ def leverage_prep(
         is_short: bool
     ) -> Tuple[float, Optional[float]]:
 
-        interest_rate = 0.0
-        isolated_liq = None
-
-        # TODO-lev: Uncomment once liq and interest merged in
         # if TradingMode == TradingMode.MARGIN:
         #     interest_rate = self.exchange.get_interest_rate(
         #         pair=pair,
         #         open_rate=open_rate,
         #         is_short=is_short
         #     )
-
-        #     if self.collateral_type == Collateral.ISOLATED:
-
-        #         isolated_liq = liquidation_price(
-        #             exchange_name=self.exchange.name,
-        #             trading_mode=self.trading_mode,
-        #             open_rate=open_rate,
-        #             amount=amount,
-        #             leverage=leverage,
-        #             is_short=is_short
-        #         )
-
-        return interest_rate, isolated_liq
+        if self.trading_mode == TradingMode.SPOT:
+            return (0.0, None)
+        elif (
+            self.collateral_type == Collateral.ISOLATED and
+            self.trading_mode == TradingMode.FUTURES
+        ):
+            isolated_liq = self.exchange.get_liquidation_price(
+                pair=pair,
+                open_rate=open_rate,
+                is_short=is_short,
+                position=amount,
+                wallet_balance=(amount * open_rate)/leverage,  # TODO: Update for cross
```

Comment:
I'm not sure i would leave this "TODO" here.

in this branch of the If - there'll be no cross (that'll be in the "else") - and if we start supporting Cross - i don't think this will be the only place to touch anyway, so maybe, this can be a comment in the else (like - should this be updated, make sure wallet-balance is calculated correctly).